core/src/main/web/app/mainapp/mainapp.js

```diff
@@ -616,8 +641,18 @@
               disconnected = true;
             },
             onReconnected: function() {
-              bkSessionManager.reconnectEvaluators();
-              disconnected = false;
+              bkSessionManager.isSessionValid()
```

Comment:
Why would it return 503? This code is evaluated after the client was successfully reconnected to the server...specifically, to the new server instance that was started by marathon after the original instance crashed.